packages/pwa-kit-react-sdk/src/ssr/universal/components/storefront-preview/refetch-data.jsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import {useHistory, useLocation} from 'react-router-dom'
+import {useQueryClient} from '@tanstack/react-query'
+import LoadingSpinner from '../loading-spinner'
+
+// For good UX, show loading spinner long enough for users to see
+const LOADING_SPINNER_DURATION = 1000
+
+const RefetchData = () => {
+    const history = useHistory()
+    const location = useLocation()
+    const queryClient = useQueryClient()
+
+    useEffect(() => {
+        queryClient.invalidateQueries()
```

Comment:
This is an `async` method. How long does it usually take to execute? Would it make sense to do `await invalidateQueries()` and then reload, rather than using a timeout?

Reply:
Oh right, it is an async method. All right, let me think about this..